Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs

```diff
@@ -2320,3 +2651,648 @@ carouselView state push =
+                    _ <- push action
+                    _ <- getCurrentPosition push UpdateCurrentLocation
+                    _ <- pure $ logEvent state.data.logField "ny_user_recenter_btn_click"
+                    pure unit
+                )
+                (const $ RecenterCurrentLocation)
+            , height $ V 40
+            , width $ V 40
+            ]
+
+        ]
+    ]
+getMapDimensions :: HomeScreenState -> {height :: Length, width :: Length}
+getMapDimensions state = 
+  let mapHeight = if (any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithDriver ] && os /= "IOS" && not state.props.locateOnMap) then 
+                    V (((screenHeight unit)/ 15)*10)
+                  else if ( (suggestionViewVisibility state) && (isBannerVisible state)|| suggestionViewVisibility state) then 
+                    V (getHeightFromPercent 27)
+                  else if (isHomeScreenView state) then
+                    V (getHeightFromPercent 45)
+                  else
+                    MATCH_PARENT
+      mapWidth =  if state.props.currentStage /= HomeScreen then MATCH_PARENT else V ((screenWidth unit)-32)
+  in {height : mapHeight, width : mapWidth}
+
+suggestionViewVisibility :: HomeScreenState -> Boolean
+suggestionViewVisibility state =  ((length state.data.tripSuggestions  > 0 || length state.data.destinationSuggestions  > 0) && isHomeScreenView state)
+
+isBannerVisible :: HomeScreenState -> Boolean
+isBannerVisible state = getValueToLocalStore DISABILITY_UPDATED == "false" && state.data.config.showDisabilityBanner && isHomeScreenView state
+
+isSaveLocationsVisible :: HomeScreenState -> Boolean
+isSaveLocationsVisible state = true 
+
+suggestionsView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM (Effect Unit) w
+suggestionsView push state = 
+  linearLayout 
+    [ width MATCH_PARENT
+    , height WRAP_CONTENT
+    , orientation VERTICAL
+    , padding $ Padding 0 0 0 16
+    , margin $ Margin 8 16 8 0
+    ][ textView $
+        [ height WRAP_CONTENT
+        , width (MATCH_PARENT)
+        , text if length state.data.tripSuggestions  == 0 then (getString SUGGESTED_DESTINATION) else (getString RECENT_RIDES)
+        , color Color.black800
+        , padding $ Padding 8 0 8 0 
+        , margin $ MarginBottom 7
+        , accessibilityHint if length state.data.tripSuggestions  == 0 then "Suggested Destinations" else "Recent Rides"
+        ] <> FontStyle.subHeading1 TypoGraphy
+      , textView $
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , text if length state.data.tripSuggestions  == 0 then (getString PLACES_YOU_MIGHT_LIKE_TO_GO_TO) else (getString ONE_CLICK_BOOKING_FOR_YOUR_FAVOURITE_JOURNEYS)
+        , color Color.black600
+        , padding $ Padding 8 0 8 0 
+        , accessibilityHint if length state.data.tripSuggestions  == 0 then "Places you might like to go to." else "One click booking for your favourite journeys!"
```

Comment:
same across the file -

instead of having length state.data.tripSuggestions  == 0 check you can directly use null for arrays which returns a boolean value directly if array is null or not

Reply:
Updated